iceoryx_hoofs/include/iceoryx_hoofs/internal/cxx/vector.inl

```diff
@@ -103,9 +103,9 @@ inline vector<T, Capacity>& vector<T, Capacity>::operator=(const vector& rhs) no
         }
 
         // delete remaining elements
-        for (; i < size(); ++i)
+        while (i < size())
```

Comment:
What do you think about using the same approach as we used for the stack.
1. Implement `clearFrom(startPosition)`
2. Use `clearFrom(i)` in move and copy
3. Use `clearFrom(0)` in `clear()`
4. Check if there is any other place in the vector where we could use this

With this we would remove some code duplication and furthermore reduce the likely-hood of future bugs. Just assume that again there is some issue with the destruction order and the bug-fixer misses a code snippet to fix since it occurs on multiple places.